llama_index/vector_stores/google/generativeai/base.py

```diff
@@ -0,0 +1,394 @@
+"""Google Generative AI Vector Store.
+
+The GenAI Semantic Retriever API is a managed end-to-end service that allows
+developers to create a corpus of documents to perform semantic search on
+related passages given a user query. For more information visit:
+https://developers.generativeai.google/guide
+"""
+
+import logging
+import uuid
+from typing import Any, Dict, List, Optional, Sequence, cast
+
+from llama_index.bridge.pydantic import BaseModel, Field, PrivateAttr  # type: ignore
+from llama_index.indices.service_context import ServiceContext
+from llama_index.schema import BaseNode, RelatedNodeInfo, TextNode
+from llama_index.vector_stores.types import (
+    BasePydanticVectorStore,
+    MetadataFilters,
+    VectorStoreQuery,
+    VectorStoreQueryResult,
+)
+
+_logger = logging.getLogger(__name__)
+_import_err_msg = "`google.generativeai` package not found, please run `pip install google-generativeai`"
+_default_doc_id = "default-doc"
+
+
+google_service_context = ServiceContext.from_defaults(
+    # Avoids instantiating OpenAI as the default model.
+    llm=None,
+    # Avoids instantiating HuggingFace as the default model.
+    embed_model=None,
+)
+"""Google GenerativeAI service context.
+
+Use this to provide the correct service context for `GoogleVectorStore`.
+
+See the docstring for `GoogleVectorStore` for usage example.
+"""
+
+
+class NoSuchCorpusException(Exception):
+    def __init__(self, *, corpus_id: str) -> None:
+        super().__init__(f"No such corpus {corpus_id} found")
+
+
+class GoogleVectorStore(BasePydanticVectorStore):
```

Comment:
I think I might have brought this up previously, but since this neither stores vectors or uses vectors for retrieval, I don't think we need to include this, as the abstraction doesn't quite make sense 🤔

Reply:
Yea that makes sense to me. There will be some initial waste in llama-index though, since we always calculate embeddings (I'm not 100% sure if there's a way to disable it)